File: Controllers/MemberController.cpp
#include "MemberController.h"
#include <iostream>
using namespace std;

MemberController::MemberController(Database& db) : db(db) {}

void MemberController::getAll() {
    cout << "\n--- ALL MEMBERS ---\n";
    db.loadMembers();
}

void MemberController::getOne(int memberID) {
    cout << "\n--- MEMBER DETAILS ---\n";
    db.getMemberByID(memberID);
}

void MemberController::create(string name, string email, int planID, int trainerID) {
    Member m;
    m.setMember(0, name, email);
    db.saveMember(m, planID, trainerID);
    cout << "✔ Member '" << name << "' registered successfully.\n";
}

void MemberController::update(int memberID, string name, string email, int planID, int trainerID) {
    db.updateMember(memberID, name, email, planID, trainerID);
    cout << "✔ Member ID " << memberID << " updated successfully.\n";
}

void MemberController::remove(int memberID) {
    db.deleteMember(memberID);
    cout << "✔ Member ID " << memberID << " deleted.\n";
}

File: Controllers/MemberController.js
// MemberController.js
// Replace `members` array with backend API calls later

let members = [];

const MemberController = {

    getAll() {
        return members;
    },

    getOne(memberID) {
        return members.find(m => m.memberID === memberID) || null;
    },

    create(memberID, name, email, planID, trainerID) {
        const exists = members.some(m => m.memberID === memberID);
        if (exists) {
            alert("Member ID already exists!");
            return false;
        }
        members.push({ memberID, name, email, planID, trainerID });
        return true;
    },

    update(memberID, name, email, planID, trainerID) {
        const index = members.findIndex(m => m.memberID === memberID);
        if (index === -1) {
            alert("Member not found!");
            return false;
        }
        members[index] = { memberID, name, email, planID, trainerID };
        return true;
    },

    remove(memberID) {
        const index = members.findIndex(m => m.memberID === memberID);
        if (index === -1) {
            alert("Member not found!");
            return false;
        }
        members.splice(index, 1);
        return true;
    },

    search(name) {
        return members.filter(m => m.name.toLowerCase() === name.toLowerCase());
    }
};